### PR TITLE
Add totalizers

### PR DESCRIPTION
**Pull request by kevinch**
Live: http://10.1.12.74:6060/#/Components/Display/Table

screen shot 2019-02-28 at 13 43 32

### CODE REVIEW

File: react/components/Table/Totalizers.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React, { Fragment, PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+import TotalizerLabel from './totalizers/TotalizerLabel'
+import TotalizerValue from './totalizers/TotalizerValue'
+import TotalizerIcon from './totalizers/TotalizerIcon'
+
+const BORDER_COLOR = 'b--light-gray'
```
Comment (klzns) [inline react/components/Table/Totalizers.js]: Use muted instead

File: react/components/Table/Totalizers.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+const TOTALIZERS_CLASSES = `w-100 flex flex-column flex-row-ns ba br3 mb5 ${BORDER_COLOR}`
+const TOTALIZER_CLASSES = `flex flex-column flex-auto pa4 ${BORDER_COLOR}`
+
+class Totalizers extends PureComponent {
+  render() {
+    const { items } = this.props
+
+    if (items.length > 0) {
```
Comment (klzns) [inline react/components/Table/Totalizers.js]: Invert this if and make it an early return

```js
if (item.length === 0) {
  return null
}
```

It should make it easier to read

File: react/components/Table/Totalizers.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      return (
+        <div id="totalizers" className={TOTALIZERS_CLASSES}>
```
Comment (klzns) [inline react/components/Table/Totalizers.js]: If used twice in the page the id will clash. Why is it needed?

File: react/components/Table/Totalizers.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          {items.map((item, i) => {
+            const IS_NOT_LAST = items.length > 1 && i < items.length - 1
+            const EXTRA_BORDER = IS_NOT_LAST ? 'bb bb-0-ns br-ns' : ''
+
+            return (
+              <div className={`${TOTALIZER_CLASSES} ${EXTRA_BORDER}`} key={i}>
```
Comment (klzns) [inline react/components/Table/Totalizers.js]: The key could be the item.label

File: react/components/Table/Totalizers.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        </div>
+      )
+    }
+
+    return null
+  }
+}
+
+Totalizers.propTypes = {
+  items: PropTypes.array.isRequired,
```
Comment (klzns) [inline react/components/Table/Totalizers.js]: A shape here would be good

File: react/components/Table/index.js [hunk 6]
```diff
@@ -248,6 +254,8 @@ Table.propTypes = {
     textOf: PropTypes.string,
     totalItems: PropTypes.number,
   }),
+  /** Totalizers property  */
+  totalizers: PropTypes.array,
```
Comment (klzns) [inline react/components/Table/index.js]: A shape here would be good

File: react/components/Table/totalizers/TotalizerIcon.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+const CIRCLE_DIAMETER = '2.4rem'
+
+class TotalizerIcon extends PureComponent {
+  render() {
+    const {
+      item: { icon, iconBackgroundColor },
```
Comment (klzns) [inline react/components/Table/totalizers/TotalizerIcon.js]: There's no iconBackgroundColor in shape of prop.

Also I think it's better as it is in the shape, limiting the option of colors to success and error

File: react/components/Table/totalizers/TotalizerLabel.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+class TotalizerLabel extends PureComponent {
```
Comment (klzns) [inline react/components/Table/totalizers/TotalizerLabel.js]: I think this could be a function component

File: react/components/Table/totalizers/TotalizerValue.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import config from 'vtex-tachyons/config.json'
+
+import Spinner from '../../Spinner'
+
+class TotalizerValue extends PureComponent {
```
Comment (klzns) [inline react/components/Table/totalizers/TotalizerValue.js]: I think this could be a function component

File: react/components/Table/totalizers/TotalizerValue.js [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    if (value) {
+      return <div className="f4 fw5 c-on-base">{value}</div>
+    }
+
+    return null
+  }
+}
+
+TotalizerValue.propTypes = {
+  item: PropTypes.object,
```
Comment (klzns) [inline react/components/Table/totalizers/TotalizerValue.js]: Define a shape

File: react/components/Table/totalizers/TotalizerValue.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    const {
+      item: { value, isLoading },
+    } = this.props
+
+    if (isLoading) {
+      return (
+        <div className="mt2">
+          <Spinner
+            size="14"
+            color={config.semanticColors.background['base--inverted']}
```
Comment (klzns) [inline react/components/Table/totalizers/TotalizerValue.js]: Use class to color the icon

File: react/components/Table/totalizers/TotalizerValue.js [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+          />
+        </div>
+      )
+    }
+
+    if (value) {
```
Comment (klzns) [inline react/components/Table/totalizers/TotalizerValue.js]: Invert this, the last case should be the default case.

```js
if (value == null) {
  return null
}
```

Also it should test if it's null or undefined like above, because if the value is zero it would return null.

File: react/components/Table/Totalizers.js (hunk 3 above)
Comment (kevinch) [reply react/components/Table/Totalizers.js]: Good point and not useful.

File: react/components/Table/Totalizers.js (hunk 4 above)
Comment (kevinch) [reply react/components/Table/Totalizers.js]: No problem with spaces in the string?

File: react/components/Table/totalizers/TotalizerIcon.js (hunk 7 above)
Comment (kevinch) [reply react/components/Table/totalizers/TotalizerIcon.js]: I've discussed it with @rsimoens and we will let it "open".

File: react/components/Table/totalizers/TotalizerValue.js (hunk 11 above)
Comment (kevinch) [reply react/components/Table/totalizers/TotalizerValue.js]: How so?

File: react/components/Table/Totalizers.js (hunk 4 above)
Comment (klzns) [reply react/components/Table/Totalizers.js]: Nope

File: react/components/Table/totalizers/TotalizerValue.js (hunk 11 above)
Comment (klzns) [reply react/components/Table/totalizers/TotalizerValue.js]:
```js
<div className="mt2 c-muted-5">
  <Spinner size="14" color="currentColor">
</div>
```
